### PR TITLE
Add KIVY_BINARY_DEPS and tests.

### PR DESCRIPTION
**Pull request by matham**
This adds KIVY_BINARY_DEPS env variable, which if present will add the contents of that path to kivy. This can be used when building e.g. a wheel where we need to add dlls (sdl, gstreamer, etc.) to the dist. One can then set the variable with the path containing the dlls.

### CODE REVIEW

File: setup.py [hunk 1]
```diff
@@ -68,6 +69,18 @@ def pkgconfig(*packages, **kw):
     return kw
 
 
+def copy_files(src, dst):
```
Comment (tito) [inline setup.py]: What about shutil.copytree ?

File: setup.py [hunk 2]
```diff
@@ -259,6 +273,11 @@ def build_extensions(self):
             for e in self.extensions:
                 e.extra_link_args += ['-lm']
 
+        if kivy_binary_deps is not None and isdir(kivy_binary_deps):
```
Comment (dessant) [inline setup.py]: `if kivy_binary_deps and isdir(kivy_binary_deps)`? 👅

File: setup.py (hunk 1 above)
Comment (matham) [inline setup.py]: @tito the reason is that copytree insists on the target directories must not exist, but here it definitely exists. There are similar issues if the subdirectories already exist. the copy_files method doesn't mind existing directories.

**Comment by matham** [comment]
Ok, added to use setuptools only if env variable is set.

**Comment by dessant** [comment]
We could document this var and possibly `KIVY_USE_SETUPTOOLS` in http://kivy.org/docs/guide/environment.html. Thanks for your packaging work (✿◠‿◠).

**Pull request merged**